src/swupd/hash.go

```diff
@@ -7,11 +7,34 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-	"syscall"
 )
 
+// #include <unistd.h>
+// #include <stdlib.h>
+// #include <stdint.h>
+// #include <sys/stat.h>
+// uint64_t get_ifmt() { return S_IFMT; }
+// uint64_t get_ifreg() { return S_IFREG; }
+// uint64_t get_ifdir() { return S_IFDIR; }
+// uint64_t get_iflnk() { return S_IFLNK; }
+import "C"
+import "unsafe"
+
 type hashval int
 
+// Defined as Go types to ensure that we can do bit operations with it.
+var s_ifmt C.uint64_t
+var s_ifreg C.uint64_t
+var s_ifdir C.uint64_t
+var s_iflnk C.uint64_t
```

Comment:
We have those constants in syscall package, that have definitions per target system.

`syscall/ztypes_windows.go:	S_IFMT   = 0x1f000
syscall/zerrors_linux_arm.go:	S_IFMT                           = 0xf000
syscall/ztypes_freebsd_386.go:	S_IFMT   = 0xf000
syscall/zerrors_linux_ppc64le.go:	S_IFMT                           = 0xf000
syscall/zerrors_darwin_arm.go:	S_IFMT                      = 0xf000
syscall/syscall_nacl.go:	S_IFMT        = 0000370000
syscall/ztypes_freebsd_amd64.go:	S_IFMT   = 0xf000
syscall/zerrors_linux_mips64.go:	S_IFMT                           = 0xf000
syscall/zerrors_linux_arm64.go:	S_IFMT                           = 0xf000
syscall/ztypes_solaris_amd64.go:	S_IFMT   = 0xf000
syscall/types_solaris.go:	S_IFMT   = C.S_IFMT
syscall/ztypes_openbsd_amd64.go:	S_IFMT   = 0xf000
syscall/ztypes_openbsd_arm.go:	S_IFMT   = 0xf000
`